### PR TITLE
odhcpd: fix dependency for IPV6

### PR DESCRIPTION
**Pull request by rosysong**
Signed-off-by: Rosy Song <rosysong@rosinson.com>

### CODE REVIEW

**Comment by jow-** [comment]
NAK, this will prevent using the DHCPv4 part of odhcpd when IPv6 is disabled.

File: package/network/services/odhcpd/Makefile [hunk 1]
```diff
@@ -32,7 +32,7 @@ define Package/odhcpd/default
   SECTION:=net
   CATEGORY:=Network
   TITLE:=OpenWrt DHCPv6(-PD)/RA Server & Relay
-  DEPENDS:=+libubox +libuci +libubus +libnl-tiny
+  DEPENDS:=+libubox +libuci +libubus +libnl-tiny @IPV6
```
Comment (dedeckeh) [inline package/network/services/odhcpd/Makefile]: NAK odhcpd has DHCPv4 support so it cannot solely depend on the config symbol IPv6

**Comment by rosysong** [comment]
When I drop CONFIG_IPV6 that I found the odhcpd can not startup and  in syslog you can see below output:.
"daemon.err odhcpd[xxx1]: socket(AF_INET6) Address family not support in protocol"
"daemon.err odhcpd[xxx2]: socket(AF_INET6) Address family not support in protocol"
"daemon.err odhcpd[xxx3]: socket(AF_INET6) Address family not support in protocol"
"daemon.err odhcpd[xxx4]: socket(AF_INET6) Address family not support in protocol"
"daemon.err odhcpd[xxx5]: socket(AF_INET6) Address family not support in protocol"

**Comment by rosysong** [comment]
@jow- @dedeckeh  uhttpd do have the same problem except it can be startup.

**Comment by dedeckeh** [comment]
@rosysong I will fix this in odhcpd

**Comment by lucize** [comment]
@rosysong nginx also, but on it  maybe there can be a an if os.execute() for /proc/sys/net/ipv6 , but that's on package feed anyway. but for nginx-luci would be nice if we have this

**Comment by rosysong** [comment]
@jow- @dedeckeh @lucize  Thank you for your replies, the only reason I pushed these IPV6 relevant PRs is that I want to keep OpenWrt always works fine whatever it support **ipv4-only** or **ipv4 with ipv6**, in the meanwhile make these PRs do not have any bad affects on the packages' building and running. So that we can build **a more embedded friendly, stable and lightweight** firmware for our corresponding machine and network environment.

**Comment by dedeckeh** [comment]
Closing as this would prevent using DHCPv4 part of odhcpd.
Further as mentioned this needs to be fixed in odhcpd

**Pull request closed without merging**